app_launcher.py

```diff
@@ -31,6 +33,12 @@ def main(input_json_path):
 
     # Change environment variables
     env = data.get("env") or {}
+
+    # Add AYON_PID_FILE environment variable if pid_file is specified
+    pid_file_path = data.get("pid_file")
```

Comment:
Shouldn't we just create the pid file? Why should it be defined in the source json?

Reply:
And why we do not just use this as source for existing `pid_file`?

Reply:
> Also the reason why the temp file is defined somewhere else is that it is much more easier to do it in python than in c++ - and we might find that we need to change how those files are created. And in that case it is better to do it at one place (in ayon-applications).

Reasonable.

> this is for an eventual shell script launched by this, not for the app_launcher itself. If pid_file is passed to the json from upstream logic, it will set it (the path to it) as environment variable to be passed to launched shell script.

Could we check if the executable is a shell script? Just trying to avoid `time.sleep` if it is not necessary.